PROTO_tests/tests/dtypes_test.py

```diff
@@ -0,0 +1,185 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+from arkouda import dtypes
+
+"""
+DtypesTest encapsulates arkouda dtypes module methods
+"""
+
+SUPPORTED_NP_DTYPES = [bool, int, float, str, np.int64, np.float64, np.uint8, np.uint64, np.str_]
+
+
+class TestDTypes:
+    @pytest.mark.parametrize("dtype", SUPPORTED_NP_DTYPES)
+    def test_check_np_dtype(self, dtype):
+        """
+        Tests dtypes.check_np_dtype method
+
+        :return: None
+        :raise: AssertionError if 1.. test cases fail
+        """
+
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype(np.dtype(np.int16))
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype("np.str")
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype("bigint")
+
+    def test_translate_np_dtype(self):
+        """
+        Tests dtypes.translate_np_dtype method
+
+        :return: None
+        :raise: AssertionError if 1.. test cases fail
+        """
+        d_tuple = dtypes.translate_np_dtype(np.dtype(np.bool_))
+        assert 1 == d_tuple[1]
+        assert "bool" == d_tuple[0]
+        d_tuple = dtypes.translate_np_dtype(np.dtype(bool))
+        assert 1 == d_tuple[1]
+        assert "bool" == d_tuple[0]
+        d_tuple = dtypes.translate_np_dtype(np.dtype(np.int64))
+        assert 8 == d_tuple[1]
+        assert "int" == d_tuple[0]
+        d_tuple = dtypes.translate_np_dtype(np.dtype(np.float64))
+        assert 8 == d_tuple[1]
+        assert "float" == d_tuple[0]
+        d_tuple = dtypes.translate_np_dtype(np.dtype(np.uint8))
+        assert 1 == d_tuple[1]
+        assert "uint" == d_tuple[0]
+        d_tuple = dtypes.translate_np_dtype(np.dtype(np.str_))
+        assert 0 == d_tuple[1]
+        assert "str" == d_tuple[0]
+        d_tuple = dtypes.translate_np_dtype(np.dtype(str))
+        assert 0 == d_tuple[1]
+        assert "str" == d_tuple[0]
+
+        with pytest.raises(TypeError):
+            dtypes.check_np_dtype(np.dtype(np.int16))
+        with pytest.raises(TypeError):
+            dtypes.translate_np_dtype("np.str")
+
+    def test_resolve_scalar_dtype(self):
+        """
+        Tests dtypes.resolve_scalar_dtype method
+
+        :return: None
+        :raise: AssertionError if 1.. test cases fail
+        """
+        assert "bool" == dtypes.resolve_scalar_dtype(True)
+        assert "int64" == dtypes.resolve_scalar_dtype(1)
+        assert "float64" == dtypes.resolve_scalar_dtype(float(0.0))
+        assert "str" == dtypes.resolve_scalar_dtype("test")
+        assert "int64" == dtypes.resolve_scalar_dtype(np.int64(1))
+        assert "<class 'list'>" == dtypes.resolve_scalar_dtype([1])
```

Comment:
I think these and a lot of other tests can be parameterized using some dictionaries.

Reply:
Updated